hazelcast/src/main/java/com/hazelcast/replicatedmap/merge/ReplicatedMapMergePolicy.java

```diff
@@ -21,25 +21,27 @@
 import java.io.Serializable;
 
 /**
- * A policy for merging replicated maps after a split-brain was detected and the different network partitions need
- * to be merged.
+ * Policy for merging replicated map entries after a split-brain has been healed.
  *
  * @see com.hazelcast.replicatedmap.merge.HigherHitsMapMergePolicy
- * @see com.hazelcast.replicatedmap.merge.PutIfAbsentMapMergePolicy
  * @see com.hazelcast.replicatedmap.merge.LatestUpdateMapMergePolicy
  * @see com.hazelcast.replicatedmap.merge.PassThroughMergePolicy
+ * @see com.hazelcast.replicatedmap.merge.PutIfAbsentMapMergePolicy
  */
 public interface ReplicatedMapMergePolicy extends Serializable {
 
     /**
-     * Returns the value of the entry after the merge
-     * of entries with the same key.
-     * You should consider the case where existingEntry's value is null.
+     * Selects one of the merging and existing map entries to be merged.
```

Comment:
I know all merge policy interfaces say something like `Selects one of the merging and existing entries to be merged`. I am wondering, do we really need to select one of those two? Can we create a new entry and return it?
This comment applies to the other interfaces as well.

Reply:
Actually all `merge()` methods have to return the `getValue()` of one of the two given entries (or `null` if the entry should be removed). So you should be able to generate an arbitrary `value` as well, which will be used for the merged `key` then. For the new `SplitBrainMergePolicy` I'm using generics to make clear, that you have to return the `value`, not one of the entries.